Add hub-spoke trust config, root CA monitoring, and kubeconfig guidance (#773)
openshift-kni/telco-reference: telco-core/configuration/reference-crs/optional/cert-manager/README.md

- `certManagerOperatorgroup.yaml` - Creates the OperatorGroup for cert-manager
- `certManagerSubscription.yaml` - Installs the OpenShift cert-manager operator

### Certificate Issuers

> - `certManagerClusterIssuer.yaml` - **ACME issuer** using Let's Encrypt with DNS-01 challenge (reference recommendation)

Reviewer: Any ACME issuer will work with this
```suggestion
- `certManagerClusterIssuer.yaml` - **ACME issuer** with DNS-01 challenge (reference recommendation)
```

Author: Done — removed "Let's Encrypt", now reads "ACME issuer with DNS-01 challenge".

1. **ClusterIssuer** (`certManagerClusterIssuer.yaml`):
   - Update `email` with your contact email
   - Configure the appropriate DNS provider for DNS-01 challenge (example shows Route53)
   - Reference pre-created Secrets for DNS provider credentials via `secretRef` — do not commit credentials in manifests

   > **Note:** Other issuer types (e.g., CA issuer for disconnected environments with existing PKI) are allowable.

>    > Users may configure their own ClusterIssuer; no reference CR is provided for non-ACME issuers.

Reviewer: ```suggestion
   > Users may configure their own ClusterIssuer; Currently only ACME issuer is provided in the reference.
```

Author: Done — updated to "currently only ACME issuer is provided in the reference."

3. Deploy the ClusterIssuer
4. Deploy the Certificate resources
5. Wait for certificates to be issued and secrets created

> 6. **(Custom CA only)** Update kubeconfig to trust the new root CA (see "Kubeconfig Trust" section below)

Reviewer: I think this line needs to be removed

Author: Done — removed the step. Kubeconfig trust applies to any non-publicly-trusted issuer, not just custom CA.

- API Server is using the certificate: Test HTTPS connection to API endpoint
- Ingress is using the certificate: Test HTTPS connection to any route

## Important: Kubeconfig Trust After API Server Cert Replacement

> > **Note:** When using a custom CA issuer, you must complete this kubeconfig update

Reviewer: Does this step need to be done for ACME issuers as well? ie by adding the root CA from the issuer?

Author: Yes, you'll need the root CA PEM from any non-publicly-trusted issuer. Reframed the section to apply generically — changed "custom CA issuer" to "non-publicly-trusted issuer" throughout.

> **Warning:** When cert-manager replaces the API server certificate with one signed by a non-publicly-trusted CA,
> existing kubeconfig files become invalid. The embedded `certificate-authority-data` still references
> the original cluster CA and cannot verify the new certificate. All `oc` and API client commands
> will fail with `x509: certificate signed by unknown authority`.

### Updating kubeconfig

1. Extract the new root CA certificate:
   ```bash
   oc get secret root-ca-secret -n cert-manager -o jsonpath='{.data.tls\.crt}' | base64 -d > /tmp/root-ca.crt
   ```

2. Update your kubeconfig to trust the new CA:
   ```bash
   oc config set-cluster $(oc config current-context | cut -d/ -f2) \
     --certificate-authority=/tmp/root-ca.crt --embed-certs
   ```

3. Verify connectivity:
   ```bash
   oc cluster-info
   ```

### Best practice for PKI environments

Generate a root CA once and use it as the root for your PKI (the ACME issuer or CA issuer your clusters will use). Add the root CA PEM to `/etc/pki/ca-trust/source/anchors/` on your workstation and run `update-ca-trust`. All certificates issued from that root CA will then be trusted without per-cluster kubeconfig updates.

> 

Reviewer 2: /etc/pki/ca-trust/source/anchors
is the place in red hat distros (RHEL, CentOS, Fedora)
in debian/ubuntu, its
/usr/bin/update-ca-trust

Author: Thanks!  I'll update the README.

> 

Reviewer 2: but more importantly, oc/kubectl does not fall back to os level trust store if the kubeconfig include certificate information, so extra step is needed to clear older kubeconfigs

> (self-signed or internal), existing kubeconfig files become invalid. The embedded
> `certificate-authority-data` still references the original cluster CA and cannot verify the new
> certificate. All `oc` and API client commands will fail with `x509: certificate signed by unknown authority`.

### Updating kubeconfig

1. Extract the new root CA certificate:
   ```bash
   oc get secret root-ca-secret -n cert-manager -o jsonpath='{.data.ca\.crt}' | base64 -d > /tmp/root-ca.crt
   ```

2. Update your kubeconfig to trust the new CA:
   ```bash
   oc config set-cluster $(oc config current-context | cut -d/ -f2) \
     --certificate-authority=/tmp/root-ca.crt --embed-certs
   ```

3. Verify connectivity:
   ```bash
   oc cluster-info
   ```

### Best practice for PKI environments

Generate a root CA once and use it as the root for your PKI (the ACME issuer or CA issuer your clusters will use). Add the root CA PEM to `/etc/pki/ca-trust/source/anchors/` on your workstation and run `update-ca-trust`. All certificates issued from that root CA will then be trusted without per-cluster kubeconfig updates.

> 

Author: Good catch. Added a Note callout at the top of the section clarifying this is a required pre-deployment step for Options B & C, and added a kubeconfig update step to the Deployment Order section in all three profile READMEs.
